test/Microsoft.ML.Tests/Scenarios/OvaTest.cs

```diff
@@ -0,0 +1,148 @@
+using Xunit;
+
+namespace Microsoft.ML.Scenarios
+{
+    public partial class ScenariosTests
+    {
+        [Fact]
+        public void OvaLogisticRegression()
+        {
+            string dataPath = GetDataPath("iris.txt");
+
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext(seed: 1);
+            var reader = new TextLoader(mlContext, new TextLoader.Arguments()
+            {
+                Column = new[]
+                        {
+                            new TextLoader.Column("Label", DataKind.R4, 0),
+                            new TextLoader.Column("Features", DataKind.R4, new [] { new TextLoader.Range(1, 4) }),
+                        }
+            });
+
+            // Data
+            var data = reader.Read(GetDataPath(dataPath));
+
+            // Pipeline
+            var pipeline = new Ova(
+                mlContext, 
+                new LogisticRegression(mlContext, "Label", "Features"),
+                useProbabilities: false);
+
+            var model = pipeline.Fit(data);
+            var predictions = model.Transform(data);
+
+            // Metrics
+            var metrics = mlContext.MulticlassClassification.Evaluate(predictions);
+            Assert.True(metrics.AccuracyMicro > 0.94);
+        }
+
+        [Fact]
+        public void OvaAveragedPerceptron()
+        {
+            string dataPath = GetDataPath("iris.txt");
+
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext(seed: 1);
+            var reader = new TextLoader(mlContext, new TextLoader.Arguments()
+            {
+                Column = new[]
+                        {
+                            new TextLoader.Column("Label", DataKind.R4, 0),
+                            new TextLoader.Column("Features", DataKind.R4, new [] { new TextLoader.Range(1, 4) }),
+                        }
+            });
+
+            // Data
+            var data = reader.Read(GetDataPath(dataPath));
+
+            // Pipeline
+            var pipeline = new Ova(
+                mlContext,
+                new AveragedPerceptronTrainer(mlContext, "Label", "Features", advancedSettings: s => { s.Shuffle = true;  s.Calibrator = null; }),
+                useProbabilities: false);
+
+            var model = pipeline.Fit(data);
+            var predictions = model.Transform(data);
+
+            // Metrics
+            var metrics = mlContext.MulticlassClassification.Evaluate(predictions);
+            Assert.True(metrics.AccuracyMicro > 0.71);
+        }
+
+        [Fact]
+        public void OvaFastTree()
+        {
+            string dataPath = GetDataPath("iris.txt");
```

Comment:
>iris.txt" [](start = 43, length = 9)

There is a TestDataset class with definitions for all datasets; if you wanted to use that rather than typing the string. In case we change files around there is only one place to change.

Reply:
Will keep those changes out of this PR, so I can get this functionality in and get going with other possible issues with OVA.

Will look into usages of TestDataset class, and incorporate that in future changes.

---
In reply to: [233720420](https://github.com/dotnet/machinelearning/pull/1574#discussion_r233720420) [](ancestors = 233720420)